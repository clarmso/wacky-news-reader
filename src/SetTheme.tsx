import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import {
  Grid,
  Radio,
  RadioGroup,
  FormControl,
  FormLabel,
  FormControlLabel,
} from "@material-ui/core";

import { set90sMode, setLightMode, setDarkMode } from "./connect/actions";
import { WackyState } from "./connect/reducer";

export const themeChoices = {
  LIGHT: "light",
  DARK: "dark",
  NINETYS: "90s",
};

const SetTheme: React.FC = () => {
  const theme = useSelector((state: WackyState) => state.themeChoice);
  const [themeChoice, setThemeChoice] = useState(theme);

  const dispatch = useDispatch();

  const handleChangeTheme = (event: React.ChangeEvent<HTMLInputElement>) => {
    const selected = event.target.value;
    switch (selected) {
      case themeChoices.NINETYS:
        dispatch(set90sMode());
        setThemeChoice(themeChoices.NINETYS);
        break;
      case themeChoices.DARK:
        dispatch(setDarkMode());
        setThemeChoice(themeChoices.DARK);
        break;
      case themeChoices.LIGHT:
        dispatch(setLightMode());
        setThemeChoice(themeChoices.LIGHT);
        break;
      default:
        dispatch(setLightMode());
        setThemeChoice(themeChoices.LIGHT);
    }
  };

  return (
    <Grid item xs>
      <FormControl>
        <FormLabel component="legend">Mode</FormLabel>
        <RadioGroup
          name="modes"
          value={themeChoice}
          onChange={handleChangeTheme}
          row
        >
          <FormControlLabel
            value={themeChoices.LIGHT}
            control={<Radio color="primary" />}
            label="Light"
            data-cy="light-mode"
          />
          <FormControlLabel
            value={themeChoices.DARK}
            control={<Radio color="primary" />}
            label="Dark"
            data-cy="dark-mode"
          />
          <FormControlLabel
            value={themeChoices.NINETYS}
            control={<Radio color="primary" />}
            label="90s ðŸŽ‰"
            data-cy="90s-mode"
          />
        </RadioGroup>
      </FormControl>
    </Grid>
  );
};

export default SetTheme;
