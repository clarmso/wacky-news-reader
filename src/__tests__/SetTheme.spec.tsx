import { Provider } from "react-redux";
import { createStore } from "redux";
import { render } from "@testing-library/react";
import SetTheme from "../SetTheme";
import wackyReducer, { initialState } from "../connect/reducer";

it("renders with initial state", () => {
  const store = createStore(wackyReducer, initialState);
  const { asFragment } = render(
    <Provider store={store}>
      <SetTheme />
    </Provider>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});
