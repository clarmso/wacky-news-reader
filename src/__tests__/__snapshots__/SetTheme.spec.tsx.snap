// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with initial state 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <div
        class="MuiFormGroup-root MuiFormGroup-row"
        role="radiogroup"
      >
        <label
          class="MuiFormControlLabel-root"
          data-cy="light-mode"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-4"
                name="modes"
                type="radio"
                value="light"
              />
              <div
                class="PrivateRadioButtonIcon-root-5 PrivateRadioButtonIcon-checked-7"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Light Mode
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
          data-cy="dark-mode"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                name="modes"
                type="radio"
                value="dark"
              />
              <div
                class="PrivateRadioButtonIcon-root-5"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Dark Mode
          </span>
        </label>
        <label
          class="MuiFormControlLabel-root"
          data-cy="90s-mode"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root MuiRadio-colorPrimary MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-4"
                name="modes"
                type="radio"
                value="90s"
              />
              <div
                class="PrivateRadioButtonIcon-root-5"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-6"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                  />
                </svg>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            90s Mode
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
